Gems/ScriptCanvas/Code/Editor/View/Windows/MainWindow.cpp

```diff
@@ -4246,13 +4260,56 @@ namespace ScriptCanvasEditor
             RemoveSystemTickAction(SystemTickActionFlag::CloseNextTabAction);
             CloseNextTab();
         }
+    }
 
-        if (m_systemTickActions == 0)
+    void MainWindow::ClearSaveAttempt()
+    {
+        m_saveAttemptInProgress = false;
+
+        if (!m_systemTickActions)
         {
             AZ::SystemTickBus::Handler::BusDisconnect();
         }
     }
 
+    bool MainWindow::EvaluateSaveAttempt()
+    {
+        const AZ::s64 k_saveAttemptSeconds = 20;
+
+        if (m_saveAttemptInProgress)
+        {
+            auto saveDuration = AZStd::chrono::seconds(AZStd::chrono::system_clock::now() - m_saveAttemptTime).count();
+
+            if (saveDuration > k_saveAttemptSeconds)
+            {
+                WarnOnFailedSaveAttempt();
+            }
+        }
+
+        return m_forceCloseInProgress;
+    }
+
+    void MainWindow::MarkSaveAttempt()
+    {
+        m_saveAttemptInProgress = true;
+        m_saveAttemptTime = AZStd::chrono::system_clock::now();
+
+        if (!AZ::SystemTickBus::Handler::BusIsConnected())
+        {
+            AZ::SystemTickBus::Handler::BusConnect();
+        }
+    }
+
+    void MainWindow::WarnOnFailedSaveAttempt()
+    {
+        m_forceCloseInProgress = true;
+        QMessageBox::critical(this, QString(), QObject::tr
+        ("The ScriptCanvas Editor has encountered an external bug which prevents it from tracking the file state.<br><br>"
+        "Likely the Asset Processor has crashed. ScriptCanvas files may have saved successfully, but the O3DE Engine and Asset Processor should be restarted before continuing work."));
+        AZ::SystemTickBus::Handler::BusDisconnect();
+        qobject_cast<QWidget*>(parent())->close();
```

Comment:
Instead of closing the widget directly, it may be necessary to call:
`QtViewPaneManager::instance()->ClosePane(LyViewPane::ScriptCanvas);`